esp/espcommon/espcommon.cpp

```diff
@@ -59,8 +59,13 @@ ESPCOMMON_API std::shared_ptr<hpccMetrics::ScaledHistogramMetric> registerServic
 {
     std::string metricName(processName);
     metricName.append(".").append(serviceName).append(".").append(methodName);
-    auto no_ = std::remove(metricName.begin(), metricName.end(), '_');
-    metricName.erase(no_, metricName.end());
+
+    // Remove unwanted characters from new metric name
+    constexpr char removeChars[] = "_-* ";
```

Comment:
minor: Where was this list derived from how about ! : or other characters which will be rejected later.

Reply:
@ghalliday 
We could add more characters here, however this method is specific to creation of a profiling metric for ESP service methods. The intent is to remove any characters the code cannot control from sources where some characters are allowed, but are forbidden by the metrics framework. The sources in this case are the SCM files and the config itself (name of the ESP process). 

For other uses of the framework, the component should be setting the metric name and ensuring it does not have illegal characters. In these cases, it might be the right thing to reject the metric lest the component owner thinks a metric has a character when the final registered metric does not. I can see the need for a separate function for cleaning a string that came from the config that could be used by components when building metric names,

Certainly we can take the approach that all metric names get scrubbed of illegal characters and add more to this list, but it would grow quite long since the regex for valid name essentially only allows upper and lower case letters, numbers, and periods (for hierarchical naming)

Thoughts?

Reply:
squash the commits and open a new issue if we see any real life problems.